awswrangler/athena/_utils.py

```diff
@@ -1144,3 +1144,94 @@ def get_query_execution(query_execution_id: str, boto3_session: Optional[boto3.S
+        Boto3 Session. The default boto3 session will be used if boto3_session receive None.
+
+    Returns
+    -------
+    List[str]
+        List of query execution IDs.
+
+    Examples
+    --------
+    >>> import awswrangler as wr
+    >>> res = wr.athena.list_query_executions(workgroup='workgroup-name')
+
+    """
+    client_athena: boto3.client = _utils.client(service_name="athena", session=boto3_session)
+    kwargs: Dict[str, Any] = {"base": 1}
+    if workgroup:
+        kwargs["WorkGroup"] = workgroup
+    query_list: List[str] = []
+    response: Dict[str, Any] = _utils.try_it(
+        f=client_athena.list_query_executions,
+        ex=botocore.exceptions.ClientError,
+        ex_code="ThrottlingException",
+        max_num_tries=5,
+        **kwargs,
+    )
+    query_list += response["QueryExecutionIds"]
+    while "NextToken" in response:
+        kwargs["NextToken"] = response["NextToken"]
+        response = _utils.try_it(
+            f=client_athena.list_query_executions,
+            ex=botocore.exceptions.ClientError,
+            ex_code="ThrottlingException",
+            max_num_tries=5,
+            **kwargs,
+        )
+        query_list += response["QueryExecutionIds"]
+    return query_list
+
+
+def get_query_executions(
+    query_execution_ids: List[str], boto3_session: Optional[boto3.Session] = None
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """From specified query execution IDs, return DataFrames of query execution details.
+
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/athena.html#Athena.Client.batch_get_query_execution
+
+    Parameters
+    ----------
+    query_execution_ids : List[str]
+        Athena query execution IDs.
+    boto3_session : boto3.Session(), optional
+        Boto3 Session. The default boto3 session will be used if boto3_session receive None.
+
+    Returns
+    -------
+    DataFrame
+        DataFrame contain information about a query execution.
+
+    DataFrame
+        DataFrame contain information about the query executions that failed to run.
+
+    Examples
+    --------
+    >>> import awswrangler as wr
+    >>> query_executions_df, unprocessed_query_executions_df = wr.athena.get_query_executions(
+            query_execution_ids=['query-execution-id','query-execution-id1']
+        )
+    """
+    chunked_size: int = 50
+    query_executions: List[Dict[str, Any]] = []
+    unprocessed_query_execution: List[Dict[str, str]] = []
+    client_athena: boto3.client = _utils.client(service_name="athena", session=boto3_session)
+    for i in range(0, len(query_execution_ids), chunked_size):
+        response = client_athena.batch_get_query_execution(QueryExecutionIds=query_execution_ids[i : i + chunked_size])
+        query_executions += response["QueryExecutions"]
+        unprocessed_query_execution += response["UnprocessedQueryExecutionIds"]
+
+    return pd.json_normalize(query_executions), pd.json_normalize(unprocessed_query_execution)
```

Comment:
Instead of two dataframes, perhaps we can consider having a single one with an additional column indicating the state of the query execution?

Reply:
@jaidisido 
I misunderstood the boto3 explanation on `UnprocessedQueryExecutionIds`, thus the doc string for `get_query_executions` is wrong .`UnprocessedQueryExecutionIds` actually contains query ids that the IAM user did not have access to when they called `BatchGetQueryExecution` (https://docs.aws.amazon.com/athena/latest/ug/workgroups-troubleshooting.html). 
Still, I think that merging two dataframes into one would make it hard for the user to know if there are some query_id that they don't have access to.